core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -104,6 +109,8 @@
     private final BindVariableSetter strSetter = this::setStrBindVariable;
     private final BindVariableSetter noopSetter = this::setNoopBindVariable;
     private final ObjList<ColumnAppender> columnAppenders = new ObjList<>();
+    private final ObjStack<IntList> bindVarTypesStack = new ObjStack<>(16);
+    private final ObjStack<NamedStatementWrapper> namedStatementWrapperStack = new ObjStack<>(16);
```

Comment:
we should have `WeakObjectPool` here. It has array dequeue underneath but does the null checking and object instantiation behind the scenes.